mantidproject/mantid
# Fix Q Range in ReflectometryReductionOne algorithm

It was found that an incorrect Q-Range was given in the IvsQ workspace returned from the ReflectometryReductionOne algorithm. We now rotate the source position in the instrument before the Conversion between IvsLambda and IvsQ is performed.

The workflow diagram of ReflectometryReductionOne now contains a `Rotate Source` step before we convert from Lamda to Q.

**To Test**
- Here is a script to compare Qmin/Qmax from a workspace, to the Qmin/Qmax from the equation
- the `dref` file can be found [here](https://slack-files.com/T02J4MMEW-F0GM248KW-7cf8ec9f38)
- Load `dref.nxs` into mantid

``` python
from mantid.simpleapi import *
from numpy import sin, pi

dref = mtd['dref']

IvsQ, IvsLam, thetaOut=ReflectometryReductionOneAuto(dref,AnalysisMode="MultiDetectorAnalysis",ProcessingInstructions="390-418", ThetaIn="0.5037",
    WavelengthMin=2.2,CorrectionAlgorithm='None',StrictSpectrumChecking=False,CorrectDetectorPositions=True)

Lam_1 = IvsLam.getItem(0)

minLam =  Lam_1.getXDimension().getMinimum()
maxLam = Lam_1.getXDimension().getMaximum()

print "min Lam: ", minLam

print "max Lam: ", maxLam

theta = 0.5037

print "min Q from EQ: ", (4 * pi * sin(theta * pi / 180)) / maxLam
print "max Q from EQ: ", (4 * pi * sin(theta * pi / 180)) / minLam

Q_1 = IvsQ.getItem(0)

print "min Q: ", Q_1.getXDimension().getMinimum()
print "max Q: ", Q_1.getXDimension().getMaximum()
```
- After running the script, ensure that both QMins are equal, as well as both QMaxs

**Release Notes**
Once this is merged I will add a section in the release notes associated with this change

Fixes #14575
